.bazelci/presubmit.yml

```diff
@@ -614,6 +614,12 @@ tasks:
   #   working_directory: examples/zig_cross_compiling
   #   build_targets:
   #     - "//..."
+  nix_cross_compiling:
+    name: Nix cross compiling test
+    platform: nixos
+    working_directory: examples/nix_cross_compiling
+    build_targets:
+      - "//:nix_cross_compiling"
```

Comment:
why not `//...`?
You should also add `test_targets`
```suggestion
      - "//..."
    test_targets:
      - "//..."
```

Reply:
While I was working on the next part to try to upstream the `@platforms//os:nixos` (instead of the constraint being defined in the example here), I deleted the code and everything still built just fine.  Evidently that's because if `target_compatible_with` isn't being satisfied, it gets silently excluded from the `//...` glob.  Hence explicitly enumerating them to catch if a platform goes missing.  It's definitely not ideal though, so maybe there's a better way to catch it?

![image](https://github.com/bazelbuild/rules_rust/assets/13792812/eb56eec6-dde5-461c-8c29-51a71722f33c)

Reply:
To repro on this branch, comment off:
```bzl
    # render_config = render_config(
    #     platforms_template = "@//bazel/cargo/platforms:{triple}",
    # ),
```

Then `//...` still passes presumably because the `filegroup()` gets transitively disabled:
```
> bazel build //...
INFO: Analyzed 15 targets (6 packages loaded, 9 targets configured).
INFO: Found 15 targets...
INFO: Elapsed time: 0.274s, Critical Path: 0.02s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
```
But:
```
> bazel build //:nix_cross_compiling                       
ERROR: Target //:nix_cross_compiling is incompatible and cannot be built, but was explicitly requested.
Dependency chain:
    //:nix_cross_compiling (47a001)
    //:rust_binary_aarch64-linux-android (47a001)
    //rust_binary:rust_binary (101ecc)
    @crate_index//:tokio (101ecc)
    @crate_index__tokio-1.34.0//:tokio (101ecc)
    @crate_index__tokio-macros-2.2.0//:tokio_macros (6cf94f)   <-- target platform (//bazel/platforms:host) didn't satisfy constraint @platforms//:incompatible
INFO: Elapsed time: 0.867s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (24 packages loaded, 606 targets configured)
```

Reply:
I would recommend adding a [build_test](https://github.com/bazelbuild/bazel-skylib/blob/main/docs/build_test_doc.md) target for all the things you want to ensure are built. But I'd rather avoid having custom settings in the presubmit definition, that way anyone can navigate to that part of the workspace and run `bazel test //...` to make sure everything works. Ideally all bazel jobs would be covered by a single `bazel test //...`.

Reply:
Looks like that has the same issue unfortunately 😞
```
> bazel test //... 
INFO: Analyzed 27 targets (0 packages loaded, 0 targets configured).
INFO: Found 26 targets and 1 test target...
INFO: Elapsed time: 0.249s, Critical Path: 0.00s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
//:nix_cross_compiling                                                  SKIPPED

Executed 0 out of 1 test: 1 was skipped.
```
Maybe we should revisit https://github.com/bazelbuild/rules_rust/pull/1976 to see if we can come up with a better way to indicate that the generated build configs don't support a particular target triple given subtle behavior of `target_compatible_with`?

Reply:
Ah dang, makes sense though

Reply:
I'm hoping in real projects that this is less of an issue.  People can still get tripped up with a `build //...` passing when it shouldn't but it would hopefully be immediately apparent when an expected artifact is suddenly missing 😅